.github/workflows/InstallRunAndArtifact.yml

```diff
@@ -0,0 +1,285 @@
+        run: |
+          function Show-Tree ($path, $indent = "") {
+              Get-ChildItem -LiteralPath $path | ForEach-Object {
+                  Write-Host "$indent|- $($_.Name)"
+                  if ($_.PSIsContainer) {
+                      Show-Tree -path $_.FullName -indent ("$indent   ")
+                  }
+              }
+          }
+          $workspace = Get-Location
+          Write-Host "Workspace directory tree for: $workspace"
+          Show-Tree $workspace
+       
+      - name: Microvinette with installed CMD.exe
+        run: |
+          $dataDir = "$(Get-Location)\MetaMorpheus\MetaMorpheus\Test\TestData"
+          & "C:\Program Files\MetaMorpheus\CMD.exe" --test -o "$dataDir\exe_output" -v minimal
+          if ($LASTEXITCODE -ne 0) {
+            Write-Host "❌ CMD.exe search failed with exit code $LASTEXITCODE" -ForegroundColor Red
+            exit $LASTEXITCODE
+          }
+          
+      - name: Generate Default TOMLs with installed CMD.dll
+        run: |
+          dotnet "C:\Program Files\MetaMorpheus\CMD.dll" -g -o "$(Get-Location)\MetaMorpheus\EngineLayer\Data"
+
+      - name: Run search with installed CMD.dll
+        run: |
+          $dataDir = "$(Get-Location)\MetaMorpheus\EngineLayer\Data"
+          dotnet "C:\Program Files\MetaMorpheus\CMD.dll" -s "$dataDir\SmallCalibratible_Yeast.mzML" -d "$dataDir\SmallYeast.fasta" -t "$dataDir\SearchTask.toml" -o "$dataDir\dll_output" -v minimal
+          if ($LASTEXITCODE -ne 0) {
+            Write-Host "❌ CMD.dll search failed with exit code $LASTEXITCODE" -ForegroundColor Red
+            exit $LASTEXITCODE
+          }
+          
+  test_build:
+    name: Search with Built
+    runs-on: windows-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.204
+
+      - name: Restore dependencies
+        run: dotnet restore ./MetaMorpheus/MetaMorpheus.sln
+
+      - name: Build solution
+        run: dotnet msbuild -v:minimal -p:Configuration=Release -p:UseSharedCompilation=false ./MetaMorpheus/
+        
+      - name: Generate Default TOMLs
+        run: |
+          dotnet "$(Get-Location)\MetaMorpheus\CMD\bin\Release\net8.0\CMD.dll" -g -o "$(Get-Location)\MetaMorpheus\CMD\bin\Release\net8.0\Data"
+
+      - name: Run search with built CMD.dll
+        run: |
+          $dataDir = "$(Get-Location)\MetaMorpheus\CMD\bin\Release\net8.0\Data"
+          dotnet "$(Get-Location)\MetaMorpheus\CMD\bin\Release\net8.0\CMD.dll" -s "$dataDir\SmallCalibratible_Yeast.mzML" -d "$dataDir\SmallYeast.fasta" -t "$dataDir\SearchTask.toml" -o "$dataDir\built_dll_output" -v minimal
+          if ($LASTEXITCODE -ne 0) {
+            Write-Host "❌ Built CMD.dll search failed with exit code $LASTEXITCODE" -ForegroundColor Red
+            exit $LASTEXITCODE
+          }
+
+  ## Installs and builds. Verifies that the same dlls exist
```

Comment:
What's the problem you've encountered that this task solves? An explanation here might help keep track of the goals here.

Reply:
We have forgotten to add new DLLs to the installer several times during my time here. The result is us needing to scramble to update the release or put out a new one. This test hopefully ensures that does not happen again

Reply:
Oh wow yeah I remember that now. This is a cool addition!